sigar_freebsd.go

```diff
@@ -111,3 +111,48 @@ func parseCpuStat(self *Cpu, line string) error {
 	self.Idle, _ = strtoull(fields[4])
 	return nil
 }
+
+func (self *Mem) Get() error {
+	val := C.uint32_t(0)
+	sc := C.size_t(4)
+
+	name := C.CString("vm.stats.vm.v_page_count")
+	_, err := C.sysctlbyname(name, unsafe.Pointer(&val), &sc, nil, 0)
+	C.free(unsafe.Pointer(name))
+	if err != nil {
+		return err
+	}
+	pagecount := uint64(val)
+
+	name = C.CString("vm.stats.vm.v_page_size")
+	_, err = C.sysctlbyname(name, unsafe.Pointer(&val), &sc, nil, 0)
+	C.free(unsafe.Pointer(name))
+	if err != nil {
+		return err
+	}
+	pagesize := uint64(val)
+
+	name = C.CString("vm.stats.vm.v_free_count")
+	_, err = C.sysctlbyname(name, unsafe.Pointer(&val), &sc, nil, 0)
+	C.free(unsafe.Pointer(name))
+	if err != nil {
+		return err
+	}
+	self.Free = uint64(val)
```

Comment:
Is this value the number of pages? `Free` is expected to be in bytes, we should probably multiply here by pagesize.

Reply:
it's done later down:

self.Used = self.Total - (self.Free * pagesize)

Reply:
But I mean to store the size in bytes in self.Free, i.e:
```
self.Free = uint64(val) * pagesize
```

`self.Free` is the value that will be used at the end for the free memory metrics.

Reply:
gotcha. fixed

Reply:
Thanks, it looks good now :)